src/search/landmarks/exploration.cc

```diff
@@ -39,19 +35,25 @@ Exploration::Exploration(const TaskProxy &task_proxy)
     // Build propositions.
     for (VariableProxy var : task_proxy.get_variables()) {
         int var_id = var.get_id();
-        propositions.push_back(vector<ExProposition>(var.get_domain_size()));
+        propositions.push_back(vector<Proposition>(var.get_domain_size()));
         for (int value = 0; value < var.get_domain_size(); ++value) {
             propositions[var_id][value].fact = FactPair(var_id, value);
         }
     }
 
-    // Build goal propositions.
-    for (FactProxy goal_fact : task_proxy.get_goals()) {
-        int var_id = goal_fact.get_variable().get_id();
-        int value = goal_fact.get_value();
-        goal_propositions.push_back(&propositions[var_id][value]);
-        termination_propositions.push_back(&propositions[var_id][value]);
+    /*
+      Reserve vector size unary operators. This is needed because we
+      cross-reference to the memory address of elements of the vector while
+      building it; meaning a resize would invalidate all references.
+    */
```

Comment:
It would probably be better to not use raw pointers to vector entries at all and for example use IDs instead.  Maybe a good follow-up issue?

Reply:
Using IDs can be better or worse than using pointers, it's hard to predict. I would only create a follow-up issue if you intend to work on it immediately, since I would say for now it's at most a "possibly nice to have".